### PR TITLE
Support impl blocks in proc-macro frontend

### PR DESCRIPTION
**Pull request by jplatte**
Part of #1257.

### CODE REVIEW

**Comment by jplatte** [comment]
Ready for review! 🎉

**Comment by jplatte** [comment]
Ah, it looks like we need to generate extra scaffolding code for destructors. I hope the force-push I just did works around that, so this PR can be merged as-is instead of growing more in terms of complexity and code size.

File: uniffi_macros/src/export/scaffolding.rs [hunk 1]
```diff
@@ -3,81 +3,169 @@
...
+            quote! {
+                #ffi_converter::try_lift(this).unwrap_or_else(|err| {
+                    ::std::panic!("Failed to convert arg 'self': {}", err)
+                }).
             }
         }
-    }
+        _ => {
+            params_args.extend(collect_params(&sig.inputs, RECEIVER_ERROR));
+            quote! { #self_ident:: }
```
Comment (bendk) [inline uniffi_macros/src/export/scaffolding.rs:70]: It's very nice how this adds support for static functions on the scaffolding side, but I don't think the support is there on the bindings side yet.  Maybe this should be commented out and replaced with a panic for now.

**Review by bendk — COMMENTED** [review]
This is great progress.  I just had a couple comments/questions.

File: uniffi_macros/src/export/scaffolding.rs (hunk 1 above)
Comment (jplatte) [reply uniffi_macros/src/export/scaffolding.rs:75]: Right. Adressed in latest commit.

**Review by bendk — APPROVED** [review]

**Pull request merged**